### ISSUE TITLE
Is this Repo still actively supported?

### ISSUE THREAD

**Issue by def1149** [sabeechen/hassio-google-drive-backup#1124]
I haven't seen a lot of action by the repo owner  @sabeechen in quite a while
Is this repo dying?

**Comment by sevorl** [comment]
Check out HA 2025.1 which features new Backup functionality. I haven't tested this yet, but it seems this addon is no longer necessary. https://www.home-assistant.io/blog/2025/01/03/3-2-1-backup/

**Comment by Steve2017** [comment]
> Check out HA 2025.1 which features new Backup functionality. I haven't tested this yet, but it seems this addon is no longer necessary. https://www.home-assistant.io/blog/2025/01/03/3-2-1-backup/

You might want to read the comments in the HA community forum first. There is a lot of unhappiness with the way it's executed... especially the encryption. Also the cloud backup offered by Nabu Casa only stores one version.

**Comment by flashadams** [comment]
Another concern I have with the way it's going is handing over backups to a cloud provider (nabu) not of my choice. Until they work the kinks out I don't think they should have pushed that new functionality in 2025.1

**Comment by def1149** [comment]
The new HA backup was designed to be extendable to other cloud systems and I saw that someone has already created an integration to Google Drive and is preparing the PR.

**Comment by gfn256** [comment]
I have forever used this add-on, and would like to take this opportunity of thanking @sabeechen for an absolutely excellent & robust add-on, in my opinion one of the best at least of those that I personally use. So a great thanks, & I sincerely hope his great work continues.

I have also been contemplating how this add-on will continue to function after the 2025.1 update which incorporates its' own scheduling & encryption system.

So far; I have not configured the new core backup features at all, & have still successfully managed to make a manual backup with the add-on, so I am guessing that the add-on's own scheduling will work as usual.

My main concern is; if I do go ahead & setup the new core backup features (I don't really need/want the encryption but I would like the ability to have a synced copy on the Nabu Casa cloud), will this add-on still operate as usual on schedule? Also; will the backups still be pruned in the normal fashion? Will these backups now be encrypted?

I'm not 100% sure this is the right place for raising these questions - so I will probably also post this on the HA core site.

**Comment by def1149** [comment]
Encryption is mandatory at this point if you enable the new HA backup. 
If you don't enable the new backup the Google backup will continue to work as it does. Nothing has been said at this time that is planned to break Google's backup.

**Comment by 1liminal1** [comment]
So just to be clear, updating to 2025.01 will not effect our current backup workflow with this addon?

**Comment by gfn256** [comment]
I'm still wondering what will happen if I setup the backup with encryption (as is required AFAIK) what will happen to my regular scheduled GDBs; will they also start being encrypted?
I'm inclined to believe they won't from this [issue/thread](https://github.com/home-assistant/core/issues/134716) I opened, and frenck kindly replied shortly & then closed the issue, but I must say I'm not 100% sure, (& possibly nor is frenck?).

**Comment by def1149** [comment]
This is a Google Backup after updating to 2025.1.0. I did not enable the new HA automatic backup.
AFAIK If you enable the new HA automatic backup, ALL your backups will be encrypted, Google and HA, because Google Backup calls the HA backup process.

![image](https://github.com/user-attachments/assets/ca57e451-5ef6-46b2-896d-093aa53cb4a4)

**Comment by gfn256** [comment]
> AFAIK If you enable the new HA automatic backup, ALL your backups will be encrypted, Google and HA, because Google Backup calls the HA backup process.

I have my doubts - but maybe you/someone can confirm after an actual test. I won't be configuring the core backup until I have firm evidence.

**Comment by hille721** [comment]
This addon continues to work without any problems together with the new HA backup solution (as they also stated in their blog).
The backups from this addon are still unencrypted as before and will show up as manual backups in the new HA backup site:

![image](https://github.com/user-attachments/assets/06b24334-0053-413a-98f8-ff025c184d02)

![image](https://github.com/user-attachments/assets/6c900ba6-af91-4eb8-911b-5f94cd8747c6)

**Comment by gfn256** [comment]
> The backups from this addon are still unencrypted as before and will show up as manual backups in the new HA backup site:

I see your images, but have you already configured (encryption key etc.) in the new core backup in HA?

**Comment by hille721** [comment]
of course as the new backup is a killer feature :)

**Comment by hille721** [comment]
![Screenshot_20250105_223601_Home Assistant.jpg](https://github.com/user-attachments/assets/223081ca-0033-4374-bf81-2675db3e8f73)

here all backups (one automatic from HA, just updated yesterday), the 5 manual ones from this Addon.

**Comment by 1liminal1** [comment]
Soooo, as long as you don't enable it, you just continue on as normal?

**Comment by Steve2017** [comment]
> This is a Google Backup after updating to 2025.1.0. I did not enable the new HA automatic backup. AFAIK If you enable the new HA automatic backup, ALL your backups will be encrypted, Google and HA, because Google Backup calls the HA backup process.
> 
I've just checked my backup from overnight and it is not encrypted. I have not activated the new backup system - just Google Drive backup. What's not clear to me is that if I enable encryption in this back-up repo (ie set password), does it encrypt both local and Google Drive?

**Comment by hille721** [comment]
> Soooo, as long as you don't enable it, you just continue on as normal?

no, you can enable it and use both in parallel without any problems. The Google Drive backups will continue to work as normal

**Comment by hille721** [comment]
As soon as this lands in HA core this Addon is really not needed anymore (especially as it seems to be unmaintained anyhow):

https://github.com/home-assistant/core/pull/134576

**Comment by gfn256** [comment]
> As soon as this lands in HA core this Addon is really not needed anymore (especially as it seems to be unmaintained anyhow):
> 
> [home-assistant/core#134576](https://github.com/home-assistant/core/pull/134576)

Thanks for this great info!

I'm only hoping that they also allow in the new core backup system encryption/non-encryption by choice & also robust scheduling mechanism. This current add-on (GDB) allows an extremely versatile scheduling mechanism (which I use). 

I'm also hoping that the retention policy will also be configurable for the Google Drive - as this add-on is.

**Comment by gfn256** [comment]
> read the announcements:

My comments above were written AFTER reading the announcements (& updating).



> Encryption is now mandatory.

Exactly why I requested the option be left to the user. (It could be defined as opt-out, but should definitely offer the option). You should read the various threads & posts from the past, of users trying to restore their systems with only encrypted backups. Also think of the scenario if one wants to just retrieve a small portion of a config file / automation etc. to copy to current setup.



> Yes, it does.

So if I want to backup every three days at 2.00 am, I can do that? You can't - read the announcements.

**Comment by def1149** [comment]
I'll continue to use Google Backup as long as it works or at least the following capabilities are implemented in HA backup:
1. Option to decrypt on download.
2. As flexible or better backup scheduling than Google Backup
3. Access to Google Drive for cloud backups. Only a single Nabu backup is unacceptable. 
Features I'd like to see:
1. Backup validation. Individual backups can be verified as not corrupt, automatically or on demand. Backups are worthless if they can't be restored.
2. Retention time set by each backup location
3. Backup events raised to HA. Start, End, Completion status

**Comment by Steve2017** [comment]
> So if I want to backup every three days at 2.00 am

This is one thing I hope they will fix shortly. I have read the explanations, but the 4:45am time set is still odd to me. The fact that there is only one Nabu Casa backup is also a weakness that I hope they will fix.

At the moment this repo is meeting all my backup needs and just works, so I see little incentive to use the HA version until it is more refined. I just don't know what they could add or how they can make it much better than this one..

**Comment by Steve2017** [comment]
I've just installed the latest HA patch. In the past I'm sure there was an option to do an HA OS back-up before installing these and the monthly update. There is nothing now. Am I dreaming that they have removed this backup feature before OS updates??

I'll just do a partial via this repo if that is the case.

**Comment by gfn256** [comment]
> Am I dreaming that they have removed this backup feature before OS updates??

I wish you were! They have taken the decision to remove the backup before update option. I haven't yet discovered what the reasoning behind this decision is - but I'm pretty sure it is linked to the new encrypted-only policy.

You can go to the [HA blog](https://community.home-assistant.io/t/2025-1-backing-up-into-2025/821339/594?page=30) to witness the steam being let off on this & the general new backup features.

**Comment by gfn256** [comment]
> At the moment this repo is meeting all my backup needs and just works, so I see little incentive to use the HA version until it is more refined. I just don't know what they could add or how they can make it much better than this one..



> I'll continue to use Google Backup as long as it works

There is something worrying I've just thought of. It is possible that even if you continue to use this add-on (or a normal service call as outlined [here](https://github.com/home-assistant/core/issues/134734#issuecomment-2576028323)) to carry on doing your backups un-encrypted, I'm not sure that the HA core is going to allow you to restore from them. This [poster](https://github.com/sabeechen/hassio-google-drive-backup/issues/1129) may be experiencing exactly this phenomenon.

**Comment by Tryfos** [comment]
The main question of the thread has not been answered. And sadly it appears that the answer is "no".

**Comment by gfn256** [comment]
> This addon continues to work without any problems together with the new HA backup solution

Have you tried restoring one of those unencrypted backups post `2025.1` ?

**Comment by Tryfos** [comment]
> > This addon continues to work without any problems together with the new HA backup solution
> 
> Have you tried restoring one of those unencrypted backups post `2025.1` ?

I agree, the addon continues to work as intended, I didn't state otherwise. Yes I've tried restoring with an unencrypted backup and it works. In fact I reverted back to 2024.12 from 2025.1.
But the main question here is whether this add-on is still actively maintained or not. The fact that it hasn't been updated for over a year now leads me to believe that it has been abandoned.

**Comment by kdelios** [comment]
> >  In fact I reverted back to 2024.12 from 2025.1.

But your 2024.12 backup was created prior 2025.1, therefore not a valid test.
You have to try to restore an unencrypted backup that created after 2025.1 with Google backup add on.

**Comment by Tryfos** [comment]
> > > In fact I reverted back to 2024.12 from 2025.1.
> 
> But your 2024.12 backup was created prior 2025.1, therefore not a valid test. You have to try to restore an unencrypted backup that created after 2025.1 with Google backup add on.

Have you tried it and it fails? I can't see a reason why. Anyway, I'm a little bit worried that the developer of this great add-on does not seem willing to maintain it any further. And for now this is my best option.

**Comment by def1149** [comment]
Try this. Download a backup created on 2025.1 with Google Backup.  
If it's not encrypted, you can browse the files and folders in the tar and view text files with an archive app.

**Comment by Tryfos** [comment]
> Try this. Download a backup created on 2025.1 with Google Backup. If it's not encrypted, you can browse the files and folders in the tar and view text files with an archive app.

I have already looked into that. All backups created by this add-on after 2025.1 are unencrypted and fully accessible. Unless the user opts to set a password in the settings and encrypt them. Otherwise they are unencrypted as before.

**Comment by hille721** [comment]
@Tryfos give it up, those guys are really resistent about believing that the Google Addon still works 😂

**Comment by nickrout** [comment]
> > > > In fact I reverted back to 2024.12 from 2025.1.
> > 
> > 
> > But your 2024.12 backup was created prior 2025.1, therefore not a valid test. You have to try to restore an unencrypted backup that created after 2025.1 with Google backup add on.
> 
> Have you tried it and it fails? I can't see a reason why. Anyway, I'm a little bit worried that the developer of this great add-on does not seem willing to maintain it any further. And for now this is my best option.

What main maintenance does it require? What is the issue you think needs fixing?

**Comment by sabeechen** [comment]
To answer the original question in this thread:

> Is this repo dying?

The short might be yes, though not because its abandoned and it isn't going to be dead soon.  Its more in an indefinite maintenance mode.  I intend to keep it "maintained" for as long as people are using it and HA doesn't change the API enough to render it inoperable.  What that means in place of the new backup features in HA ... I'm not entirely sure.  I share the frustration with how they've implemented the new backup features but its also obvious this is just their first pass and more changes are to come.  I'll continue to keep this addon working as well as I can, but I can only "react" to changes they make.

I go for long periods of time not answering issues because, honestly, I'm kind of sick of answering the same questions over and over.  95% of what people ask in this repo is answered by just reading the log output from HA and the supervisor in the issue they post.  The error is right there in plain text, its why I added that bug reporting mechanism.  I'm gonna lose it if I have to explain to someone again that the addon isn't "leaking CPU", you just have big backups that take a long time!  That could probably be considered harsh or negligent on my part, but I just don't have the same kind of time I used to.  I started this project while taking a break from work a few years ago, now I've got a full time job and some babies to care for so thats always going to come first.  AFAIK the addon has a few uncommon (and hard to track down) bugs but is broadly speaking still fully functional, even with HA 2025.1.  What I _really do_ need to do in the next few weeks it comb through the 150+ issues in this repo and see if there is anythign actionable.

> All backups created by this add-on after 2025.1 are unencrypted and fully accessible

Thankfully thats what I've seen too.  HA 2025.1 adds a lot of stuff to the native backup panel that makes it seem like backups are "always" encrypted, and in the blog post they even say `Encrypted backups by default` as a section title but backups created through the API (which is what this addon does) are still unencrypted unless you specify a password.  I imagine I'm going have a lot of confused users post issues here because of that choice of language.  Despite that there appears to be no change to the way backups work from this addon's perspective.  If some think the new automatic, encrypted, Nabu Casa based backup is sufficient for their needs, I think thats great and they should use it.  First party support like that has strong advantages.  Its good for the community that this addon isn't the only serious game in town for resilient backup.

**Comment by def1149** [comment]
Thanks for the reply. I didn't intend to sound hostile or unappreciative of your effort. 
I was concerned about losing Google Backup. 
I will continue to use Google Backup. It has been flawless and reliable. It was one of the first add-ons Integrated when I started using HA about 15 months ago. I depended heavily on restores to recover my system that I butchered countless times while climbing the HA learning curve. Dropping backups before trying stuff and knowing I had reliable recovery was very helpful. 
I will use Google Backup until HA breaks it or the HA functionality is better than the Google Backup functionality. 
Thanks for a nice, easy to use, reliable design

**Comment by Steve2017** [comment]
Thanks @sabeechen 

Your work has saved me a lot of time and given me some assurance that I can’t/won’t lose the backups. I did buy you a coffee via PayPal - not much - but you will need all you can get with the babies!

**Comment by baerakafubu** [comment]
Everything said before - thx @sabeechen for that highly valueable add-on!

**Comment by sabeechen** [comment]
No worries and I've taken no offense, given my silence its very reasonable to assume I'd died or something, thats my bad.  I'll try to keep on top of new issues better for the near future, I think I'm going to:
- Set up one of those automations that deletes stale/dead issues.
- Maybe also one of those new-fangled AI bots that auto-responds for the most common bug reports I get.  So tired of telling people to got look at the damn supervisor logs.
-  Add some language to the readme acknowledging the new first-party backup.  If that feature matures some I could easily recommend it for most use cases, it addresses _most_ of the rough edges in HA backup this addon was written to solve.  Not all of them though.

If anyone knows of any "real" problems buried among my hunderds of open bugs, @sabeechen on it to draw my eye.

**Comment by gfn256** [comment]
I'll just say it one more time sabeechen - simply awesome!

**Comment by hs6666** [comment]
interesting sabeechen, i think your situ really mirrors a very common issue around open source projects

i don't feel hopeful about open source in the long term, anecdotally there seems to be less and less of the newer younger generation taking up the open source mantel

which is a shame as i would recommend to any up and coming talented youngster to stop paying for all these courses, actively involve themselves in this type of project, take over some of the responsibilities, eventually even fork it if needed with everyone's blessing

the things they would learn over the period (not just programming but maintaining/interacting) would allow them to walk into jobs over those who have studied purely theoretically.

So beginning their career path before it became time for them to have babies and pass on the mantel to someone younger :)

**Comment by kdelios** [comment]
![Image](https://github.com/user-attachments/assets/fd63132f-62e2-4184-8eb8-e7bc67e077c9)

**Comment by gfn256** [comment]
It appears that a partial restore may not work.

See [here](https://community.home-assistant.io/t/add-on-home-assistant-google-drive-backup/107928/892).

**Comment by dm82m** [comment]
in HA core 2025.2.0 there is a lot changed in the direction of backups, so maybe @sabeechen can retire from developing this addon and use the time for family caretaking. ❤ apart from that: thank you for your great work over the last years, it was more than one time I would be screwed without having your backup solution in place!

just a short wrap up:
- google drive is now supported (and also onedrive and synology dsm)
- encryption can be turned off
- if encryption is tuned on and you download the backup through HA it is unencrypted on download
- custom scheduling of backups is possible
- retention can be set based on amount of backup files or days
- better filename handling, no cryptic name - it now describes the content it holds, i.e. _Automatic_backup_2025.2.0b5_-_2025-02-01_16.47_58275713_

there is only one feature I use from this plugin here which is not on the HA side yet:
- having a different amount on local vs cloud backups, I had 3 local and 31 cloud, but this is not that important to me

details can be found [here](https://rc.home-assistant.io/blog/2025/01/29/release-20252/).

if you want to test it out, you need to wait for final 2025.2.0 release or [switch to beta](https://www.home-assistant.io/common-tasks/os/#running-a-beta-version) and follow [this guide](https://rc.home-assistant.io/integrations/google_drive/).

**Comment by def1149** [comment]
I will now switch to Backup that is now part of HA core as it has achieved functional parity with Google Backup

@sabeechen  Thank you for your very nice backup for HA.

**Comment by Dude4Linux** [comment]
Thank you @sabeechen for a great addition to HA. You saw an obvious hole in HA's functionality and took steps to remedy the situation. You've set the standard for what a good backup system should achieve. I'm sorry that the core developers didn't offer you a path for integration and instead felt the need to reinvent the wheel. NIH syndrome I guess.
The success of HA is due in part to the efforts of independent developers such as yourself who have contributed countless hours making HA a better product. I also understand the need to bring important addons into the core so they become part of the build/test for new releases and to reduce the maintenance burden on the original developer.
Hope you enjoy your coffee's.
